api/cluster.go

```diff
@@ -86,8 +87,10 @@ func (s *Server) indexList(ctx *middleware.Context, req models.IndexList) {
 func (s *Server) getData(ctx *middleware.Context, request models.GetData) {
 	series, err := s.getTargetsLocal(request.Requests)
 	if err != nil {
+		// the only errors returned are from us catching panics, so we should treat them
+		// all as internalServerErrors
```

Comment:
a) just because it is an internal endpoint doesnt mean that every error should be classified as an internal error.  If the caller sends bad data then this endpoint definitely should return a 400 response.

b) in future this will almost certainly return errors other then internalServer error.  The obvious example is a timeout reading from cassandra resulting in a ServiceUnavailable error.